python/src/equistore/tensor.py

```diff
@@ -394,6 +460,146 @@ def property_names(self) -> List[str]:
 
+    def remove_gradients(self, remove: Optional[List[str]] = None) -> None:
+        """Remove some or all of the gradients.
+
+        :param remove:
+            which gradients should be excluded from the new tensor map.
+            If this is set to ``None`` (this is the default), all the gradients will
+            be removed.
+        """
+        from equistore.operations import remove_gradients
+
+        return remove_gradients(self, remove)
+
+    def dot(self, other: "TensorMap") -> "TensorMap":
+        """Computes the dot product with another :py:class:`TensorMap`.
+
+        The given :py:class:`TensorMap` must have the same ``keys``.
+
+        :py:class:`TensorBlocks` corresponding to the same key must have the same
+        ``properties``. The resulting :py:class:`TensorBlocks` of the dot product of
+        two :py:class:`TensorBlocks` has ``result_block.values = block1.values @
+        block2.values.T``
+
+        :param other:
+            :py:class:`TensorMap` to multiply
+
+        :return: dot product
+        """
+        from equistore.operations import dot
+
+        return dot(self, other)
+
+    def _reduce_over_samples(
+        self, samples_names: Union[List[str], str], reduction: str
+    ) -> "TensorMap":
+        from equistore.operations.reduce_over_samples import _reduce_over_samples
+
+        return _reduce_over_samples(self, samples_names, reduction)
+
+    def sum_over_samples(self, samples_names: List[str]) -> "TensorMap":
+        """Aaverage over the indices of the given ``samples_names`` .
+
+        For addiotional details refer to the reference page of
+        the :ref:`python-api-operations-sample-reduction` functions.
+
+        :param samples_names:
+            names of samples to sum over
+        """
+        return self._reduce_over_samples(samples_names, "sum")
+
+    def mean_over_samples(self, samples_names: List[str]) -> "TensorMap":
+        """Aaverage over the indices of the given ``samples_names`` .
+
+        For addiotional details refer to the reference page of
+        the :ref:`python-api-operations-sample-reduction` functions.
+
+        :param samples_names:
+            names of samples to average over
+        """
+        return self._reduce_over_samples(samples_names, "mean")
+
+    def std_over_samples(self, samples_names: List[str]) -> "TensorMap":
+        """Standard deviation over the indices of the given ``samples_names``.
+
+        For addiotional details refer to the reference page of
+        the :ref:`python-api-operations-sample-reduction` functions.
+
+        :param samples_names:
+            names of samples to perform the standart deviation over
+        """
+        return self._reduce_over_samples(samples_names, "std")
+
+    def variance_over_samples(self, samples_names: List[str]) -> "TensorMap":
+        r"""Variance over the indices of the given ``samples_names``.
+
+        For addiotional details refer to the reference page of
```

Comment:
addiotional

Reply:
Thanks @DavideTisi for always pointing out my typos 🤣